bedevere/backport.py

```diff
@@ -58,3 +63,33 @@
 
     backport_issue = await util.issue_for_PR(gh, pull_request)
     await _copy_over_labels(gh, original_issue, backport_issue)
+
+
+@router.register("pull_request", action="opened")
+@router.register("pull_request", action="edited")
+@router.register("pull_request", action="synchronize")
+async def validate_backport_pr(event, gh, *args, **kwargs):
+    """ Detect when PRs made against maintenance branch, and the title does not
+    match backport PR pattern.
+    Post a failure status check.
+    """
+    if event.data["action"] == "edited" and "title" not in event.data["changes"]:
+        return
+    pull_request = event.data["pull_request"]
+    base_branch = pull_request["base"]["ref"]
+
+    if base_branch == "master":
+        return
+
+    title = util.normalize_title(pull_request["title"],
+                                 pull_request["body"])
+    title_match = TITLE_RE.match(title)
```

Comment:
Is there a reason to worry about PRs that are only to fix a bug in a maintenance branch and has no commit against `master`? I'm thinking of the case when something was implicitly fixed by something and the need to fix a maintenance release comes up later. In that instance there wouldn't necessarily be a PR to refer to.

Reply:
I'm thinking that PR to the maintenance branch that is not a backport of another PR should be quite rare, and if happens we should be "suspicious" about it. That's why I'm checking for both [X.Y] and (GH-NNNN) here.
But I propose to make this status check optional, so we can still merge it even if the status check failed.

Reply:
Fair enough. We can see if anyone complains about the failure being a big deal on this cases and adjust accordingly.